### PR TITLE
convert many more tests to junit 4

### PR DESCRIPTION
**Pull request by grimreaper**
and clean up some tests I didn't convert in prep for the next round

### CODE REVIEW

File: code/src/testcommon/compare/WeakReferenceInequality.java [hunk 1]
```diff
@@ -24,9 +24,9 @@
 {
 
     @Override
-	public String testInequality(WeakReference<?> wr1, WeakReference<?> wr2, InequalityTester t, String location)
+	public String testInequality(WeakReference<?> t1, WeakReference<?> t2, InequalityTester t, String location)
```
Comment (thpr) [inline code/src/testcommon/compare/WeakReferenceInequality.java]: Why this change?

Comment (grimreaper) [inline code/src/testcommon/compare/WeakReferenceInequality.java]: make it match the function it is overriding
